pymatgen/symmetry/analyzer.py

```diff
@@ -1697,3 +1700,543 @@ def __str__(self):
+
+    def get_parent_spacegroup(self):
+        """
+        Returns analyzer of the parent spacegroup, i.e. the spacegroup of 
+        the nonmagnetic structure.
+
+        Returns:
+            pymatgen.symmetry.analyzer.SpacegroupAnalyzer
+        """
+        return self._spacegroupAnalyzer
+
+    def get_paramagnetic_symmetry_operations(self):
+        """
+        Returns all magnetic symmetry operations of the nonmagnetic structure, which comprises all
+        crystallographic spacegroup operations with time reversal symmetry even and odd.
+      
+        Returns:
+            list(pymatgen.core.operations.MagSymmOp): List of symmetry operations.
+        """
+        return self._paramag_symmetry_ops
+
+    def _paramag_symmetry_operations(self, cartesian=False):
+        """
+        Computes all magnetic symmetry operations of the nonmagnetic structure, which comprises all
+        crystallographic spacegroup operations with time reversal symmetry even and odd.
+        By default returns symmetry operations in fractional coordinates for a given structure,
+        but can also return symmetry operations in Cartesian coordinates.
+
+        Returns:
+            list(pymatgen.core.operations.MagSymmOp): List of symmetry operations.
+        """
+        if np.any(np.abs(self._structure._lattice.matrix - self._spacegroupAnalyzer._cell[0]) > 1e-12):
+            exit("Error in the definitions of the unit cell.")
+
+        rotation = self._spacegroupAnalyzer._space_group_data['rotations']
+        translation = self._spacegroupAnalyzer._space_group_data['translations']
+        
+        # if cell is conventional prim2cart will be conv2cart
+        prim2cart = np.transpose(self._spacegroupAnalyzer._cell[0])
+        prim2cart_inv = np.linalg.inv(prim2cart)
+        # if cell is already conventional prim2conv will be unit matrix
+        prim2conv = self._spacegroupAnalyzer._space_group_data["transformation_matrix"]
+        prim2conv_inv = np.linalg.inv(prim2conv)
+
+        symmops = []
+        for rot, trans in zip(rotation, translation):
+            if cartesian:
+                rot = prim2cart.dot(rot.dot(prim2cart_inv))
+                trans = prim2cart.dot(trans)
+            else:
+                rot = prim2conv.dot(rot.dot(prim2conv_inv)).round(8)
+                trans = prim2conv.dot(trans).round(8)
+          
+            op = SymmOp.from_rotation_and_translation(rot, trans)
+            symmops.append(op)
+        
+        paramag_symmops = [MagSymmOp.from_symmop(op, time_reversal) for op in symmops for time_reversal in [+1, -1]]
+        
+        return paramag_symmops
+
+    def get_magnetic_pointgroup(self):
+        """
+        Get magnetic pointgroup information as tuple (label, number).
+
+        Returns:
+            tuple(str, list(int)): Magnetic point group for structure.
+        """
+        MAGSPACEGROUP2MAGPOINTGROUP = os.path.join(os.path.dirname(__file__), "mspg2mpg_map.txt")
```

Comment:
We can make this a module-level variable so that the file doesn't have to be re-loaded and parsed every time. I can make a further practical suggestion here for how to structure the file, so that it's not necessary to iterate over lines individually.

Reply:
I've attached a possible alternative as a yaml file (will have to be unzipped, since GitHub doesn't allow me to upload .yaml directly).

Then, at the top of this file, we could write:

```
from monty.serialization import loadfn

MAGSPACEGROUP_TO_MAGPOINTGROUP = loadfn(os.path.join(os.path.dirname(__file__), "magspacegroup_to_magpointgroup.yaml"))
```

This will be loaded as a dictionary keyed by the OG no., so in this `get_magnetic_pointgroup` function it can just `return MAGSPACEGROUP_TO_MAGPOINTGROUP[glob_og_no]` for example.

(monty.serialization.loadfn and monty.serialization.dumpfn are convenience functions for loading and dumping Python objects, pymatgen objects etc. to JSON or YAML)

[magspacegroup_to_magpointgroup.yaml.zip](https://github.com/materialsproject/pymatgen/files/5915205/magspacegroup_to_magpointgroup.yaml.zip)